core/src/main/java/org/infinispan/configuration/cache/ClusteringConfiguration.java

```diff
@@ -36,11 +38,19 @@ static AttributeSet attributeDefinitionSet() {
          PartitionHandlingConfiguration partitionHandlingStrategy) {
       this.attributes = attributes.checkProtection();
       this.cacheMode = attributes.attribute(CACHE_MODE);
+      this.remoteTimeout = attributes.attribute(REMOTE_TIMEOUT);
+      this.invalidationBatchSize = attributes.attribute(INVALIDATION_BATCH_SIZE);
       this.hashConfiguration = hashConfiguration;
       this.l1Configuration = l1Configuration;
       this.stateTransferConfiguration = stateTransferConfiguration;
       this.syncConfiguration = syncConfiguration;
       this.partitionHandlingConfiguration  = partitionHandlingStrategy;
+
+      // Expose the true value for users of attributes().attribute(REMOTE_TIMEOUT).get()
+      // and attributes().attribute(REMOTE_TIMEOUT).addListener()
+      syncConfiguration.attributes().attribute(SyncConfiguration.REPL_TIMEOUT)
```

Comment:
I'll create a separate JIRA, but the changes have to be in this PR because `RpcManager.getSyncRpcOptions()` really needs the `remoteTimeout` listener to work -- no test actually checked the listeners before.